### PR TITLE
Check length argument is positive integer in common lattices

### LINKED ISSUE #895: [input validation] nk.graph.Chain([L]) does not fial
nk.graph.Chain([L]) should fail, but does not, and leads to failure later that it took me a while to realise...

We should validate the input to be an integer.

```python
>>> graph = nk.graph.Chain([L])
>>> sgb = graph.space_group_builder()
>>> sgb.little_group([0])
```

### PR DESCRIPTION
**Pull request by femtobit**
This resolves #895.

### CODE REVIEW

File: netket/graph/common_lattices.py [hunk 1]
```diff
@@ -162,6 +164,8 @@ def Cube(length: int, *, pbc: bool = True) -> Lattice:
         >>> print(g.n_nodes)
         1000
     """
+    if not isinstance(length, int) or length <= 0:
+        raise TypeError("Argument `length` must be a positive integer")
```
Comment (PhilipVinc) [inline netket/graph/common_lattices.py:168]: I think we do not need this because it calls hypercube?

File: netket/graph/common_lattices.py [hunk 2]
```diff
@@ -183,6 +187,8 @@ def Square(length: int, *, pbc: bool = True) -> Lattice:
         >>> print(g.n_nodes)
         100
     """
+    if not isinstance(length, int) or length <= 0:
+        raise TypeError("Argument `length` must be a positive integer")
```
Comment (PhilipVinc) [inline netket/graph/common_lattices.py:191]: as above

File: netket/graph/common_lattices.py [hunk 3]
```diff
@@ -203,6 +209,8 @@ def Chain(length: int, *, pbc: bool = True) -> Lattice:
         >>> print(g.n_nodes)
         10
     """
+    if not isinstance(length, int) or length <= 0:
+        raise TypeError("Argument `length` must be a positive integer")
```
Comment (PhilipVinc) [inline netket/graph/common_lattices.py:213]: as above

File: test/graph/test_graph.py [hunk 4]
```diff
@@ -305,6 +305,18 @@ def test_graph_wrong():
     with pytest.raises(ValueError):
         nk.graph.Graph([1, 2, 3], [1, 2, 3])
 
+    with pytest.raises(TypeError):
+        nk.graph.HyperCube([5])
```
Comment (femtobit) [inline test/graph/test_graph.py:309]:
```suggestion
        nk.graph.Hypercube([5])
```

File: netket/graph/common_lattices.py (hunk 1 above)
Comment (femtobit) [reply netket/graph/common_lattices.py]: Good point.

**Review by PhilipVinc — APPROVED** [review]
great

**Pull request merged**